### ISSUE TITLE
PHP 8.3

### ISSUE THREAD

**Issue by kickthemooon** [longxinH/xhprof#89]
![Screenshot 2024-10-18 at 08 16 56](https://github.com/user-attachments/assets/7673f683-5012-4b0b-b9eb-5a9b10bb04f7)

![Screenshot 2024-10-18 at 08 18 15](https://github.com/user-attachments/assets/8ccf9130-cdbc-4a50-ae93-e5194206b865)

![Screenshot 2024-10-18 at 08 20 05](https://github.com/user-attachments/assets/4fad3945-dbc6-44ff-8792-44d483f99abf)

I have seemingly managed to enable the extension and added the function in my index.php application entrypoint.

But, unfortunately, I'm not getting any output under /tmp/xhprof nor the directory itself.

Did anybody manage to get it working with PHP 8.3?

Is there something else I should do to enable it for FPM?

I also called it via the CLI bypassing FPM

```
env -i REDIRECT_STATUS=true \
  SCRIPT_NAME=index.php \
  SCRIPT_FILENAME=index.php \
  REQUEST_METHOD=GET \
  REQUEST_URI="/some-path" \
  SERVER_NAME="www.example.com" \
  cgi-fcgi -bind -connect 127.0.0.1:9000
```

The response comes back successfully but no profiling data under /tmp/xhprof nor the directory itself.

**Comment by Purpl3ttuce** [comment]
Hey, i just saw this and am wondering if support for php 8.4 is in place or planned anytime soon, do you know? Thanks

**Comment by longxinH** [comment]
> Hey, i just saw this and am wondering if support for php 8.4 is in place or planned anytime soon, do you know? Thanks

Of course, due to work delays, support for PHP 8.4 will be launched as soon as possible.

**Comment by kasparsd** [comment]
It works fine even with PHP 8.4 with the latest fixes from #87.

> But, unfortunately, I'm not getting any output under /tmp/xhprof nor the directory itself.

This extension doesn't output anything to that directory by default. You need to write custom code that pipes the output of `xhprof_disable()` to your desired location. See an example here:

https://github.com/longxinH/xhprof/blob/20b8caffe24cf942b9930b043aab79da6c913db8/examples/sample.php#L22-L26

That directory setting is used as a default for the included library and nothing else (which is confusing, I agree):

https://github.com/longxinH/xhprof/blob/20b8caffe24cf942b9930b043aab79da6c913db8/xhprof_lib/utils/xhprof_runs.php#L89-L95
